src/tds/challenge.c

```diff
@@ -225,6 +227,9 @@ make_lm_v2_response(const unsigned char ntlm_v2_hash[16],
 	int mac_len = 16 + client_data_len;
 	unsigned char *mac;
 
+	if (client_data == NULL	 ||  client_data_len <= 0)
+		return NULL;
+
```

Comment:
useless code

Reply:
Good point.  Perhaps more `_len` variables and arguments should be `size_t` or some other unsigned type for clarity, though.